src/deroff.rs

```diff
@@ -1060,23 +1164,56 @@ fn test_comment() {
     assert_eq!(deroffer.s, "\nworld".to_owned());
 }
 
-fn deroff_files(files: &[String]) -> std::io::Result<()> {
+fn deroff_files<P: AsRef<std::path::Path>, S: ToString>(
+    files: &[S],
```

Comment:
I think using `&[String]` or `&[&str]` is fine for me. If `ToString` is really needed, we could use `Display` but I think the original is better.

Reply:
I like your idea of putting this and things like it in it's own container, I'll make it faithful to the original and comment it out with a TODO